src/uu/skill/src/util.rs

```diff
@@ -0,0 +1,139 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::collections::HashSet;
+use uu_pgrep::process::{walk_process, ProcessInformation};
+#[cfg(target_os = "linux")]
+pub fn get_all_processes() -> Vec<ProcessInformation> {
+    walk_process().collect()
+}
+
+#[cfg(target_os = "linux")]
+pub fn filter_processes_by_pid(pids: &[i32]) -> Vec<ProcessInformation> {
+    let mut matching_pids = Vec::new();
+
+    for process in get_all_processes() {
+        if pids.iter().any(|pid| *pid == process.pid as i32) {
+            matching_pids.push(process);
+        }
+    }
+
+    matching_pids
+}
+
+#[cfg(target_os = "linux")]
+pub fn filter_processes_by_user(users: &[String]) -> Vec<ProcessInformation> {
+    let mut matching_pids = Vec::new();
+
+    for mut process in get_all_processes() {
+        if let Some(owner) = get_process_owner(&mut process) {
+            if users.iter().any(|u| *u == owner) {
+                matching_pids.push(process);
+            }
+        }
+    }
+
+    matching_pids
+}
+
+#[cfg(target_os = "linux")]
+pub fn filter_processes_by_command(commands: &[String]) -> Vec<ProcessInformation> {
+    let mut matching_processes = Vec::new();
+
+    for process in get_all_processes() {
+        let cmdline = process.cmdline.split(" ").collect::<Vec<_>>()[0];
+        let cmd_name = cmdline.split("/").last().unwrap_or(cmdline);
+        if commands.iter().any(|c| c == cmd_name) {
+            matching_processes.push(process);
+        }
+    }
+
+    matching_processes
+}
+
+#[cfg(target_os = "linux")]
+pub fn filter_processes_by_terminal(terminals: &[String]) -> Vec<ProcessInformation> {
+    let mut matching_processes = Vec::new();
+
+    for process in get_all_processes() {
+        if let Some(tty) = get_process_terminal(&process) {
+            if terminals.iter().any(|t| tty.contains(t)) {
+                matching_processes.push(process);
+            }
+        }
+    }
+
+    matching_processes
+}
+
+#[cfg(target_os = "linux")]
+pub fn get_process_owner(process: &mut ProcessInformation) -> Option<String> {
+    // let status_path = format!("/proc/{}/status", pid);
```

Comment:
I would either remove this line or add a TODO with a reason why it's commented out.